packages/renderer/src/lib/preferences/Util.ts

```diff
@@ -37,16 +44,24 @@ export interface IConnectionRestart {
 }
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function writeToTerminal(xterm: any, data: string[], colorPrefix: string): void {
+export function writeToTerminal(xterm: any, data: any[], colorPrefix: string): void {
   if (Array.isArray(data)) {
-    for (const it of data) {
-      writeMultilineString(xterm, it, colorPrefix);
-    }
-  } else {
+    writeArrayToTerminal(xterm, data, colorPrefix);
+  } else if (typeof data === 'string') {
     writeMultilineString(xterm, data, colorPrefix);
   }
 }
 
+function writeArrayToTerminal(xterm: any, data: any[], colorPrefix: string) {
+  for (const content of data) {
+    if (Array.isArray(content)) {
+      writeArrayToTerminal(xterm, content, colorPrefix);
+    } else if (typeof content === 'string') {
+      writeMultilineString(xterm, content, colorPrefix);
+    }
+  }
+}
+
```

Comment:
I would expect a separate PR for this change, including more tests due to the change of behavior

Reply:
Ok, moving it